k9mail/src/main/java/com/fsck/k9/activity/FolderList.java

```diff
@@ -615,17 +615,18 @@ private void onOpenFolder(String folder) {
         if (folder.equals(allMessagesFolderName)) {
             SearchAccount allMessagesAccount = SearchAccount.createAllMessagesAccount(this);
             MessageList.actionDisplaySearch(this, allMessagesAccount.getRelatedSearch(), false, false, true);
-            return;
-        }
-        if (folder.equals(unifiedFolderName)) {
+        } else if (folder.equals(unifiedFolderName)) {
```

Comment:
why would we extract or reuse folders detection methods here?
anyways, there is a set of application supported type of folders we know already, and when in nearest future we want to add more to support, it will be even easier to make.